doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -762,58 +683,287 @@ the conversion in hypotheses :n:`{+ @ident}`.
+   Performs the specified reduction on :n:`@term` and displays
+   the resulting term with its type. If a proof is open, :n:`@term`
+   may reference hypotheses of the selected goal.  :cmd:`Eval` is
+   a :token:`query_command`, so it may be prefixed with a goal selector.
+
+   .. cmd:: Compute @term
+
+      Evaluates :n:`@term` using the bytecode-based virtual machine.
+      It is a shortcut for :cmd:`Eval` :n:`vm_compute in @term`.
+      :cmd:`Compute` is a :token:`query_command`, so it may be prefixed
+      with a goal selector.
+
+.. _controlling-the-reduction-strategies:
+
+Controlling reduction strategies and the conversion algorithm
+-------------------------------------------------------------
+
+The commands to fine-tune the reduction strategies and the lazy conversion
+algorithm are described in this section.  Also see :ref:`Args_effect_on_unfolding`,
+which supports additional fine-tuning.
+
+.. cmd:: Opaque {+ @reference }
+
+   Marks the specified constants as :term:`opaque` so tactics won't :term:`unfold` them
+   with :term:`delta-reduction`.
+   "Constants" are items defined by commands such as :cmd:`Definition`,
+   :cmd:`Let` (with an explicit body), :cmd:`Fixpoint`, :cmd:`CoFixpoint`
+   and :cmd:`Function`.
+
+   This command accepts the :attr:`global` attribute.  By default, the scope
+   of :cmd:`Opaque` is limited to the current section or module.
+
+   :cmd:`Opaque` also affects Coq's conversion algorithm, causing
+   it to delay unfolding the specified constants as much as possible when it
+   has to check that two distinct applied constants are convertible.
+   See Section :ref:`conversion-rules`.
+
+.. cmd:: Transparent {+ @reference }
+
+   The opposite of :cmd:`Opaque`, it marks the specified constants
+   as :term:`transparent`
+   so that tactics may unfold them.  See :cmd:`Opaque` above.
+
+   This command accepts the :attr:`global` attribute.  By default, the scope
+   of :cmd:`Transparent` is limited to the current section or module.
+
+   Note that constants defined by proofs ending with :cmd:Qed: are
+   irreversibly opaque; :cmd:`Transparent` will not make them transparent.
+   This is consistent
+   with the usual mathematical practice of *proof irrelevance*: what
+   matters in a mathematical development is the sequence of lemma
+   statements, not their actual proofs. This distinguishes lemmas from
+   the usual defined constants, whose actual values are of course
+   relevant in general.
+
+   .. exn:: The reference @qualid was not found in the current environment.
+
+      There is no constant named :n:`@qualid` in the environment.
+
+.. seealso:: :ref:`applyingconversionrules`, :cmd:`Qed` and :cmd:`Defined`
+
+.. _vernac-strategy:
+
+.. cmd:: Strategy {+ @strategy_level [ {+ @reference } ] }
+
+   .. insertprodn strategy_level strategy_level_or_var
+
+   .. prodn::
+      strategy_level ::= opaque
+      | @integer
+      | expand
+      | transparent
+      strategy_level_or_var ::= @strategy_level
```

Comment:
Move the definition of `strategy_level_or_var` to the description of the `with_strategy` tactic below.